homeassistant/components/reolink/host.py

```diff
@@ -0,0 +1,176 @@
+        """Connect to Reolink host."""
+        self._api.expire_session()
+
+        if not await self._api.get_host_data():
+            return False
+
+        if self._api.mac_address is None:
+            return False
+
+        enable_onvif = None
+        enable_rtmp = None
+        enable_rtsp = None
+
+        if not self._api.onvif_enabled:
+            _LOGGER.info(
+                "ONVIF is disabled on %s, trying to enable it", self._api.nvr_name
+            )
+            enable_onvif = True
+
+        if not self._api.rtmp_enabled and self._api.protocol == "rtmp":
+            _LOGGER.info(
+                "RTMP is disabled on %s, trying to enable it", self._api.nvr_name
+            )
+            enable_rtmp = True
+        elif not self._api.rtsp_enabled and self._api.protocol == "rtsp":
+            _LOGGER.info(
+                "RTSP is disabled on %s, trying to enable it", self._api.nvr_name
+            )
+            enable_rtsp = True
+
+        if enable_onvif or enable_rtmp or enable_rtsp:
+            if not await self._api.set_net_port(
+                enable_onvif=enable_onvif,
+                enable_rtmp=enable_rtmp,
+                enable_rtsp=enable_rtsp,
+            ):
+                if enable_onvif:
+                    _LOGGER.error(
+                        "Unable to switch on ONVIF on %s. You need it to be ON to receive notifications",
+                        self._api.nvr_name,
+                    )
+
+                if enable_rtmp:
+                    _LOGGER.error(
+                        "Unable to switch on RTMP on %s. You need it to be ON",
+                        self._api.nvr_name,
+                    )
+                elif enable_rtsp:
+                    _LOGGER.error(
+                        "Unable to switch on RTSP on %s. You need it to be ON",
+                        self._api.nvr_name,
+                    )
+
+        if self._unique_id is None:
+            self._unique_id = format_mac(self._api.mac_address)
+
+        return True
+
+    async def update_states(self) -> bool:
+        """Call the API of the camera device to update the states."""
+        return await self._api.get_states()
+
+    async def disconnect(self):
+        """Disconnect from the API, so the connection will be released."""
+        try:
+            await self._api.unsubscribe_all()
+        except Exception as error:  # pylint: disable=broad-except
+            err = str(error)
+            if err:
+                _LOGGER.error(
+                    "Error while unsubscribing ONVIF events for %s: %s",
+                    self._api.nvr_name,
+                    err,
+                )
+            else:
+                _LOGGER.error(
+                    "Unknown error while unsubscribing ONVIF events for %s",
+                    self._api.nvr_name,
+                )
+
+        try:
+            await self._api.logout()
+        except Exception as error:  # pylint: disable=broad-except
```

Comment:
Please trap only expected exceptions

Reply:
I removed the broad exception and only catch the expected exceptions now.